src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -229,25 +232,96 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(Runti
         }
 
         /// <summary>
-        /// Check whether the entity name defined in runtime config only contains
-        /// characters allowed for GraphQL names.
-        /// Does not perform validation for entities which do not
+        /// Check whether the entity configuration defined in runtime config only contains characters allowed for GraphQL names
+        /// and other validations related to rest path and methods configured for the entity.
+        /// The GraphQL validation is not performed for entities which do not
         /// have GraphQL configuration: when entity.GraphQL == false or null.
         /// </summary>
         /// <seealso cref="https://spec.graphql.org/October2021/#Name"/>
-        /// <param name="entityCollection">The runtime entities to process.</param>
-        public static void ValidateEntityNamesInConfig(RuntimeEntities entityCollection)
+        /// <param name="runtimeConfig">The runtime configuration.</param>
+        public static void ValidateEntityConfiguration(RuntimeConfig runtimeConfig)
         {
-            foreach ((string _, Entity entity) in entityCollection)
+            // Stores the unique rest paths configured for different entities present in the config.
+            HashSet<string> restPathsForEntities = new();
+
+            foreach ((string entityName, Entity entity) in runtimeConfig.Entities)
             {
-                if (entity.GraphQL.Enabled)
+                if (runtimeConfig.Runtime.Rest.Enabled)
+                {
+                    // By default we assume Rest is enabled for the entity.
+                    bool isRestEnabledForEntity = true;
+
+                    // If no custom rest path is defined for the entity, we default it to the entityName.
+                    string pathForEntity = entityName;
+                    if (entity.Rest is not null)
```

Comment:
can we add the check for enabled directly in this if:

 `if (entity.Rest is not null && entity.Rest.Enabled)`

Reply:
Even if Rest is disabled we still need to perform the validations related to path done by the method `ValidateRestPathSettingsForEntity`. So, we cannot club the two conditions into one if.

Reply:
Can you add a comment in the code for "Even if Rest is disabled we still need to perform the validations related to path done by the method" and add an explanation "why?"

Reply:
why would we need to do those path validations if rest is disabled?

Reply:
I thought this is what we discussed, Ani. The user might disable rest for an entity, but the paths for entities have to be unique irrespective of that. And similarly, the path must not contain reserved characters or be empty.

Reply:
Its been a long time since we discussed this, so I don't remember what the reasoning behind this decision was. At this point, it seems unnecessary to validate if the path of an entity contains reserved characters/or is empty if the entity is disabled for rest. Do you remember the reasoning?

Reply:
The exact reasoning I am unable to recall but one argument in favour of validating is - we are validating that two entities do not have duplicate paths even if rest is disabled on both of them. Now suppose if the path had reserved characters, then it would give the impression from the exception that it was configured correctly- just that two entities had the same path configured lead to the exception, which is not the case.

Otherwise, yes. I don't feel and felt like we should be doing validations when rest is disabled. And I think the duplicate path validation should also be done only when rest is enabled on both of the entities.

Reply:
With your latest changes, isn't duplicate path validation only happening when rest is enabled for both of the entities?
line 260 `if (isRestEnabledForEntity && !restPathsForEntities.Add(pathForEntity))` here: https://github.com/Azure/data-api-builder/blob/381a893774282c9ccb061f75b318a60e782fbb2a/src/Core/Configurations/RuntimeConfigValidator.cs#L260C22-L260C22

Reply:
Doing only when rest is enabled now.